### PR TITLE
Fix a race condition which could cause a test failure on a slow CI machine

### PR DESCRIPTION
**Pull request by jbzdarkid**
## Proposed changes

Describe what this PR is trying to do.

## Type of change

- [ ] Feature work
- [ ] Bug fix
- [ ] Documentation
- [ ] Engineering change
- [X] Test
- [ ] Logging/Telemetry

## Risk

- [ ] High – Errors could cause MAJOR regression of many scenarios. (Example: new large features or high level infrastructure changes)
- [ ] Medium – Errors could cause regression of 1 or more scenarios. (Example: somewhat complex bug fixes, small new features)
- [X] Small – No issues are expected. (Example: Very small bug fixes, string changes, or configuration settings changes)

## Additional information

We saw a failure in one of our nightly builds in this test, so I am fixing it.

### CODE REVIEW

File: IdentityCore/tests/mac/MSIDMacKeychainTokenCacheTests.m [hunk 1]
```diff
@@ -724,6 +724,7 @@ - (void)testSaveAccount_whenAccountSaved_shouldSaveValidLastModInfo
     // check behavior if item had been written by a different process:
```
Comment (kaisong1990) [inline IdentityCore/tests/mac/MSIDMacKeychainTokenCacheTests.m:724]: This pull request does not update changelog.txt.

Please consider if this change would be noticeable to a partner or user and either update changelog.txt or resolve this conversation.

**Review by hieunguyenmsft — APPROVED** [review]

**Comment by jbzdarkid** [comment]
There was an unrelated CI failure...

testMSIDLRUCache_whenCallingAPIsUseThrottlingCacheWithinGCDBlocks_throttlingCacheShouldPerformOperationsWithThreadSafety, ((customLRUCache.numCacheRecords) equal to (100-customLRUCache.cacheUpdateCount)) failed: ("74") is not equal to ("75")

`/Users/travis/build/AzureAD/microsoft-authentication-library-common-for-objc/IdentityCore/tests/MSIDLRUCacheTest.m:383`

    XCTAssertEqual(customLRUCache.numCacheRecords,100-customLRUCache.cacheUpdateCount);

**Comment by hieunguyenmsft** [comment]
> There was an unrelated CI failure...
> 
> testMSIDLRUCache_whenCallingAPIsUseThrottlingCacheWithinGCDBlocks_throttlingCacheShouldPerformOperationsWithThreadSafety, ((customLRUCache.numCacheRecords) equal to (100-customLRUCache.cacheUpdateCount)) failed: ("74") is not equal to ("75")
> 
> `/Users/travis/build/AzureAD/microsoft-authentication-library-common-for-objc/IdentityCore/tests/MSIDLRUCacheTest.m:383`
> 
> ```
> XCTAssertEqual(customLRUCache.numCacheRecords,100-customLRUCache.cacheUpdateCount);
> ```
@NerevarineRule Peter fyi

**Comment by jbzdarkid** [comment]
@hieunguyenmsft It failed again on iOS, and this time it failed on macOS too.

**Comment by NerevarineRule** [comment]
@jbzdarkid  Sorry for the inconvenience. Could you try again please?

**Pull request merged**